### PR TITLE
Feature 32 balance with params

### PR DESCRIPTION
**Pull request by VeikkoVilpponen**
Chances to the api are done for now. The issue requires chances to the ui.

### CODE REVIEW

**Review by Kaitsuus — COMMENTED** [review]
To me this looks fine

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestTimeEntriesData.kt [hunk 1]
```diff
@@ -23,178 +23,181 @@ class TestTimeEntriesData {
...
-                    person = 1,
-                    task = 123,
-                    nonProjectTime = null,
-                    timeRegistered = 42,
-                    date = getSixtyDaysAgo().toString(),
-                    createdAt = getODT(getSixtyDaysAgo().atStartOfDay()),
-                    updatedAt = getODT(getSixtyDaysAgo().atStartOfDay())
-                )
+            val entries = mutableListOf(
+                    createTestTimeEntry(
```
Comment (Jdallos) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestTimeEntriesData.kt:27]: The only changes in this file are the double indentations, this should be undone.

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/PersonsTest.kt [hunk 2]
```diff
@@ -153,7 +153,9 @@ class PersonsTest: AbstractTest() {
         createTestBuilder().use { testBuilder ->
             val personTotalTimes = testBuilder.manager.persons.getPersonTotal(
                 personId = 2,
-                timespan = null
+                timespan = null,
+                before = LocalDate.now().minusDays(1).toString(),
```
Comment (Jdallos) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/PersonsTest.kt:157]: Does this set the time to an appropriate value e.g. will all timeEntries from the previous day be included?

**Review by Jdallos — CHANGES_REQUESTED** [review]

File: gradle.properties [hunk 3]
```diff
@@ -5,6 +5,6 @@ quarkusPlatformArtifactId=quarkus-universe-bom
 quarkusPluginId=io.quarkus
 quarkusPlatformGroupId=io.quarkus
 quarkusPlatformVersion=2.9.0.Final
-org.gradle.jvmargs=-Xmx2024m -XX:MaxPermSize=512m
+org.gradle.jvmargs=-Xmx2024m -XX:MaxMetaspaceSize=512m
```
Comment (villejuutila) [inline gradle.properties:8]: Could these jvmargs be removed all together?

File: src/main/kotlin/fi/metatavu/timebank/api/impl/PersonsApi.kt [hunk 4]
```diff
@@ -21,12 +22,14 @@ class PersonsApi: PersonsApi, AbstractApi() {
     @Inject
     lateinit var personsTranslator: PersonsTranslator
 
-    override suspend fun listPersonTotalTime(personId: Int, timespan: Timespan?): Response {
+    override suspend fun listPersonTotalTime(personId: Int, timespan: Timespan?, before: LocalDate?, after: LocalDate?): Response {
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/timebank/api/impl/PersonsApi.kt:25]: The submodule within this PR doesn't reference to an updated version of the spec and therefore this method doesn't seem to override anything. That is also causing the Google Cloud Build to fail.

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestTimeEntriesData.kt (hunk 1 above)
Comment (villejuutila) [reply src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestTimeEntriesData.kt:27]: What Josh said.
Is the indentation of your IntelliJ IDEA configured to 4 or 2 tabs/spaces? That can be found on the bottom right corner of the IDE.

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 3 above)
Comment (VeikkoVilpponen) [reply gradle.properties:8]: If removed jvmargs will revert to default settings of Xmx512m "-XX:MaxMetaspaceSize=256m"

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestTimeEntriesData.kt (hunk 1 above)
Comment (VeikkoVilpponen) [reply src/test/kotlin/fi/metatavu/timebank/api/test/functional/data/TestTimeEntriesData.kt]: It is 4 spaces for me.

**Review by Jdallos — APPROVED** [review]

**Pull request merged**